www/js/rover-api.js

```diff
@@ -94,11 +93,23 @@ function acceptName() {
 				saveDesign();
 				$('#nameErrorArea').empty();
 				$('a#designNameArea').text(designName);
-				$('a#downloadLink').attr("href", "saved-bds/"+designName+".xml");
-				$('a#downloadLink').attr("download", designName+".xml");
+				$('a#downloadLink').attr("onclick", "return downloadDesign(\""+designName+".xml\")");
 				$('#nameModal').foundation('reveal', 'close');
 			}
 		});
 	}
 
 }
+
+function downloadDesign(name) {
+  $.get(roverResource(['download', name]), function(response) {
+    var blob = new Blob([new XMLSerializer().serializeToString(response)]);
+    var downloadUrl = URL.createObjectURL(blob);
+    $(document.createElement('a'))
+      .attr( {'download': name, 'href': downloadUrl} )
+      .get(0)
+      .click();

```

Comment:
It's fascinating that this is the recommended way to do this. But, it doesn't actually end up looking that messy.